src/sinks/util/tcp.rs

```diff
@@ -198,7 +213,8 @@ impl Sink for TcpSink {
                 //
                 // If this returns `WouldBlock` we know the connection is still
                 // valid and the write will most likely succeed.
-                match connection.get_mut().read(&mut [0u8; 1]) {
+                let stream = connection.get_mut().get_mut().get_mut().get_mut();
```

Comment:
I feel like this line needs a type annotation to kind of pin the exact type we're expecting here, after the chain of `get_mut`s.
It seems like we want a raw TCP socket here, because we're doing wouldblock assertions below. It looks like an important detail that can be lost.

Reply:
Added types annotations for this and another 2 places.